Add retention subsystem (#55125)
zephyrproject-rtos/zephyr: doc/releases/release-notes-3.4.rst

  board-specific configuration in board Kconfig fragments in the ``boards``
  folder of the application.

> * For nRF51/nRF52-based boards, the behaviour of reset reason provided
>   to :c:func:`sys_reboot` being set in the GPREGRET register has been dropped,
>   this function will now just reboot the device without changing the register
>   contents. The new method for setting this register is by using the boot mode
>   feature of the retention subsystem, see the
>   :ref:`boot mode API <boot_mode_api>` for details. To restore the old
>   deprecated functionality of reset reason of :c:func:`sys_reboot` being set to
>   the GPREGRET register, enable
>   :kconfig:option:`CONFIG_NRF_STORE_REBOOT_TYPE_GPREGRET`.

Reviewer: ```suggestion
* On nRF51 and nRF52-based boards, the behaviour of the reset reason provided
  to :c:func:`sys_reboot` and being set in the GPREGRET register has been dropped.
  This function will now just reboot the device without changing the register
  content. The new method for setting this register uses the boot mode
  feature of the retention subsystem, see the
  :ref:`boot mode API <boot_mode_api>` for details. To restore the
  deprecated functionality, enable
  :kconfig:option:`CONFIG_NRF_STORE_REBOOT_TYPE_GPREGRET`.
```
